vivian_airflow_extensions/operators/snowflake_to_postgres_operator.py

```diff
@@ -102,29 +127,38 @@ def execute(self, context):
                 self.columns_list = self.postgres_hook.get_table_metadata(self.postgres_table, self.schema, self.include_autoincrement_keys)
                 self.metadata_retrieved = True
                 self.columns_to_update = [col for col in self.columns_list if col not in self.primary_key_columns]
-                
-            columns_to_update_string = ", ".join([f'"{col}"=excluded."{col}"' for col in self.columns_to_update])
+                self._validate_insert_only_columns()
+
+            columns_to_update_array = []
+            for column in self.columns_to_update:
+                if self.insert_only_columns is not None and column in self.insert_only_columns:
+                    line = f'"{column}"=coalesce(source."{column}", excluded."{column}")'
+                else:
+                    line = f'"{column}"=excluded."{column}"'
+                columns_to_update_array.append(line)
+            columns_to_update_string = ", ".join(columns_to_update_array)
             primary_key_columns_string = ", ".join(['"' + col + '"' for col in self.primary_key_columns])
             on_conflict_clause = f'on conflict({primary_key_columns_string}) do update set {columns_to_update_string}'
 
             if self.conditional_psql_timestamp_column is not None:
-                conditional_timestamp_clause = f'where excluded."{self.conditional_psql_timestamp_column}" >= "{self.postgres_table}"."{self.conditional_psql_timestamp_column}"'
+                conditional_timestamp_clause = f'where excluded."{self.conditional_psql_timestamp_column}" >= source."{self.conditional_psql_timestamp_column}"'
             else:
                 conditional_timestamp_clause = ''
 
         tmp_table = f'Tmp{self.postgres_table}'
 
         if self.columns_to_update is None:
-            self.insert_commands = [f'insert into "{self.postgres_table}" select * from "{tmp_table}" {on_conflict_clause};']
+            self.insert_commands = [f'insert into "{self.postgres_table}" as source select * from "{tmp_table}" {on_conflict_clause};']
```

Comment:
why are you aliasing this as source?

Reply:
to simplify the coalesce below: `coalesce(source."col", excluded."col")`

Reply:
Hmm I only see a coalesce statement above on L135